packages/binding.foreach/src/foreach.ts

```diff
@@ -281,7 +318,7 @@ export class ForEachBinding extends AsyncBindingHandler {
     if (as) {
       return v => this._contextExtensions($ctx.extend({ [as]: v }))
     } else {
-      return v => $ctx.createChildContext(v, null, ctx => this._contextExtensions(ctx))
+      return v => $ctx.createChildContext(v, undefined, ctx => this._contextExtensions(ctx))
```

Comment:
I assume our tests will cover the consequences of this, and that it's fine, but just noting that it's a somewhat semantic change in fairly fickle code.  _Probably_ ok, but noting just in case.

Reply:
It _should_ be covered. I'm always a bit unsure about this situations in JS. Is there a general rule like "undefinied is better than null" in JS/TS-Styling-Guides?

Reply:
The general rule is that they are different and fickle and all a bit insane e.g.

```
undefined == null is true // equality with type-cast
undefined === null is false // strictly inequal
null + 1 is 1
undefined + 1 is NaN

function x (def = 1) { return def }
x(null) is null
x(undefined) is 1

typeof null is "object"
typeof undefined is "undefined"
```

So it's often dangerous to change old code from `null` to `undefined` and vice versa because of these (and many more) subtle differences that may not be immediately apparent, particularly if they're not covered by tests.

If we can keep this as `null`, that'd be safer.